### ISSUE TITLE
`toRef` can pass `vue/no-dupe-keys` allowing duplicated reference in template

### ISSUE THREAD

**Issue by hanool** [vuejs/eslint-plugin-vue#3070]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 10.1.0
- **eslint-plugin-vue version:** 10.8.0
- **Vue version:** 3.5.31
- **Node version:** 24.14.1
- **Operating System:** Ubuntu24.04 (on WSL2)

**Please show your full configuration:**
I'm using default `@nuxt/eslint` module
```js
// @ts-check
import withNuxt from './.nuxt/eslint.config.mjs'

export default withNuxt(
  // Your custom configs here
)

```

**What did you do?**	

```vue
<template>
  <div>
    <!-- template will have reference for both ref foo and props.foo -->
    <p>{{foo}}</p>
  </div>
</template>
<script setup>
import { ref, toRef } from "vue"
const props = defineProps(['foo'])
// const foo = ref() // this will be errored with vue/no-dupe-keys
const foo = toRef(props, 'foo') // this passes vue/no-dupe-keys, and toRef foo.
</script>
```

**What did you expect to happen?**
`const foo = toRef(props, 'foo')` is marked as invalidation of `vue/no-dupe-keys`

**What actually happened?**
`const foo = toRef(props, 'foo')` passes `vue/no-dupe-keys`

**Repository to reproduce this issue**
[playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxlUMtqxDAM/BXhy2Zhd30vbr6h9LruIY1lYppYRnZSSsi/148tLfSmkUYzGu1iJIPiSaiES5iHhL32AMq4rRa5DP2+W6LjUDK0oWxTJf/sqDiyCwkiprXQ3BKIE+zAaC+Q6BUtHGCZFtBiW1EL7UfyMUFgChGewaB1Hl8K6u6n7Hh6O2svJTRabmRSVuvOkJtpchE+3TzDOwIyE6PJOE2QxaWnq1kDXj/wK/74NIF6SVc9L1BdfuXCECPGfwL5/EfOZjgNG5ZDkNGPWHUtcctxK2jw5pE4o1t5VHtOL45vaEmAEg==)
